### PR TITLE
chore(broker): introduce configurable step timeout

### LINKED ISSUE #3609: BrokerReprocessingTest is flaky due to not able to start broker.
**Description**

Qa tests are sometimes failing because the brokers cannot be started.

```
11:46:42.070 [] [Time-limited test] INFO  io.zeebe.broker.system - Bootstrap Broker-0 [10/11]: leader management request handler
11:46:42.074 [] [Time-limited test] DEBUG io.zeebe.broker.system - Bootstrap Broker-0 [10/11]: leader management request handler started in 3 ms
11:46:42.074 [] [Time-limited test] INFO  io.zeebe.broker.system - Bootstrap Broker-0 [11/11]: zeebe partitions
11:46:42.078 [] [Time-limited test] INFO  io.zeebe.broker.system - Bootstrap Broker-0 partitions [1/1]: partition 1
11:46:42.263 [GatewayTopologyManager] [Broker-0-zb-actors-0] DEBUG io.zeebe.gateway - Received metadata change from Broker 0, partitions {} and terms {}.
11:46:42.373 [Broker-0-ZeebePartition-1] [Broker-0-zb-actors-1] WARN  io.zeebe.logstreams - Unexpected non-empty log failed to read the last block
11:46:42.479 [io.zeebe.logstreams.impl.log.LogStreamImpl] [Broker-0-zb-actors-1] WARN  io.zeebe.logstreams - Unexpected non-empty log failed to read the last block
11:46:42.971 [io.zeebe.logstreams.impl.log.LogStreamImpl] [Broker-0-zb-actors-1] DEBUG io.zeebe.logstreams - Configured log appender back pressure at partition 0 as AppenderVegasCfg{initialLimit=1024, maxConcurrency=32768, alphaLimit=0.7, betaLimit=0.95}. Window limiting is disabled
11:46:42.984 [Broker-0-ZeebePartition-1] [Broker-0-zb-actors-1] DEBUG io.zeebe.broker.system - Removing follower partition service for partition PartitionId{id=1, group=raft-partition}
11:46:42.985 [Broker-0-ZeebePartition-1] [Broker-0-zb-actors-1] DEBUG io.zeebe.broker.system - Partition role transitioning from null to LEADER
11:46:43.059 [Broker-0-ZeebePartition-1] [Broker-0-zb-actors-1] DEBUG io.zeebe.broker.system - Installing leader partition service for partition PartitionId{id=1, group=raft-partition}
11:46:43.780 [Broker-0-ZeebePartition-1] [Broker-0-zb-actors-1] DEBUG io.zeebe.logstreams.snapshot - Available snapshots: []
11:47:12.181 [] [Time-limited test] INFO  io.zeebe.broker.system - Bootstrap Broker-0 partitions [1/1]: partition 1 failed with unexpected exception.
java.util.concurrent.TimeoutException: Timeout after: 30 SECONDS
	at io.zeebe.util.sched.future.CompletableActorFuture.get(CompletableActorFuture.java:131) ~[zeebe-util-0.22.0-SNAPSHOT.jar:0.22.0-SNAPSHOT]
	at io.zeebe.util.sched.FutureUtil.join(FutureUtil.java:32) ~[zeebe-util-0.22.0-SNAPSHOT.jar:0.22.0-SNAPSHOT]
	at io.zeebe.util.sched.future.CompletableActorFuture.join(CompletableActorFuture.java:201) ~[zeebe-util-0.22.0-SNAPSHOT.jar:0.22.0-SNAPSHOT]
	at io.zeebe.broker.Broker.scheduleActor(Broker.java:234) ~[zeebe-broker-0.22.0-SNAPSHOT.jar:0.22.0-SNAPSHOT]
	at io.zeebe.broker.Broker.lambda$partitionsStep$14(Broker.java:299) ~[zeebe-broker-0.22.0-SNAPSHOT.jar:0.22.0-SNAPSHOT]
	at io.zeebe.broker.bootstrap.StartProcess.lambda$startStepByStep$2(StartProcess.java:60) ~[zeebe-broker-0.22.0-SNAPSHOT.jar:0.22.0-SNAPSHOT]
	at io.zeebe.broker.bootstrap.StartProcess.takeDuration(StartProcess.java:88) ~[zeebe-broker-0.22.0-SNAPSHOT.jar:0.22.0-SNAPSHOT]
	at io.zeebe.broker.bootstrap.StartProcess.startStepByStep(StartProcess.java:58) ~[zeebe-broker-0.22.0-SNAPSHOT.jar:0.22.0-SNAPSHOT]
	at io.zeebe.broker.bootstrap.StartProcess.takeDuration(StartProcess.java:88) ~[zeebe-broker-0.22.0-SNAPSHOT.jar:0.22.0-SNAPSHOT]
	at io.zeebe.broker.bootstrap.StartProcess.start(StartProcess.java:43) ~[zeebe-broker-0.22.0-SNAPSHOT.jar:0.22.0-SNAPSHOT]
	at io.zeebe.broker.Broker.partitionsStep(Broker.java:303) ~[zeebe-broker-0.22.0-SNAPSHOT.jar:0.22.0-SNAPSHOT]
	at io.zeebe.broker.Broker.lambda$initStart$9(Broker.java:179) ~[zeebe-broker-0.22.0-SNAPSHOT.jar:0.22.0-SNAPSHOT]
	at io.zeebe.broker.bootstrap.StartProcess.lambda$startStepByStep$2(StartProcess.java:60) ~[zeebe-broker-0.22.0-SNAPSHOT.jar:0.22.0-SNAPSHOT]
	at io.zeebe.broker.bootstrap.StartProcess.takeDuration(StartProcess.java:88) ~[zeebe-broker-0.22.0-SNAPSHOT.jar:0.22.0-SNAPSHOT]
	at io.zeebe.broker.bootstrap.StartProcess.startStepByStep(StartProcess.java:58) ~[zeebe-broker-0.22.0-SNAPSHOT.jar:0.22.0-SNAPSHOT]
	at io.zeebe.broker.bootstrap.StartProcess.takeDuration(StartProcess.java:88) ~[zeebe-broker-0.22.0-SNAPSHOT.jar:0.22.0-SNAPSHOT]
	at io.zeebe.broker.bootstrap.StartProcess.start(StartProcess.java:43) ~[zeebe-broker-0.22.0-SNAPSHOT.jar:0.22.0-SNAPSHOT]
	at io.zeebe.broker.Broker.internalStart(Broker.java:129) ~[zeebe-broker-0.22.0-SNAPSHOT.jar:0.22.0-SNAPSHOT]
	at io.zeebe.util.LogUtil.doWithMDC(LogUtil.java:21) [zeebe-util-0.22.0-SNAPSHOT.jar:0.22.0-SNAPSHOT]
	at io.zeebe.broker.Broker.start(Broker.java:119) [zeebe-broker-0.22.0-SNAPSHOT.jar:0.22.0-SNAPSHOT]
	at io.zeebe.broker.test.EmbeddedBrokerRule.startBroker(EmbeddedBrokerRule.java:209) [zeebe-broker-0.22.0-SNAPSHOT-tests.jar:0.22.0-SNAPSHOT]
	at io.zeebe.broker.test.EmbeddedBrokerRule.before(EmbeddedBrokerRule.java:124) [zeebe-broker-0.22.0-SNAPSHOT-tests.jar:0.22.0-SNAPSHOT]
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:50) [junit-4.13.jar:4.13]
	at org.junit.rules.RunRules.evaluate(RunRules.java:20) [junit-4.13.jar:4.13]
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:288) [junit-4.13.jar:4.13]
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:282) [junit-4.13.jar:4.13]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
```

### PR DESCRIPTION
**Pull request by ChrisKujawa**
## Description

 * step timeout was limited by 30 seconds, which was often not enough - sometimes it took longer to start
 * default step timeout was increased to 5 minutes
 * it is now configurable via configuration and enviromnent variable

## Related issues

closes https://github.com/zeebe-io/zeebe/issues/3609

## Pull Request Checklist

- [X] All commit messages match our [commit message guidelines](https://github.com/zeebe-io/zeebe/blob/develop/CONTRIBUTING.md#commit-message-guidelines)
- [X] The submitting code follows our [code style](https://github.com/zeebe-io/zeebe/wiki/Code-Style)
- [X] If submitting code, please run `mvn clean install -DskipTests` locally before committing

### CODE REVIEW

File: broker/src/main/java/io/zeebe/broker/system/configuration/BrokerCfg.java [hunk 1]
```diff
@@ -23,6 +23,8 @@
   private EmbeddedGatewayCfg gateway = new EmbeddedGatewayCfg();
   private BackpressureCfg backpressure = new BackpressureCfg();
 
+  private String stepTimeout = "5m";
```
Comment (deepthidevaki) [inline broker/src/main/java/io/zeebe/broker/system/configuration/BrokerCfg.java]: Should this be exposed to users? If yes, we should also add it to the config file for documentation.

File: broker/src/main/java/io/zeebe/broker/system/configuration/EnvironmentConstants.java [hunk 2]
```diff
@@ -21,4 +21,5 @@
   public static final String ENV_CLUSTER_NAME = "ZEEBE_CLUSTER_NAME";
   public static final String ENV_EMBED_GATEWAY = "ZEEBE_EMBED_GATEWAY";
   public static final String ENV_DEBUG_EXPORTER = "ZEEBE_DEBUG";
+  public static final String ENV_STEP_TIMEOUT_EXPORTER = "ZEEBE_STEP_TIMEOUT";
```
Comment (deepthidevaki) [inline broker/src/main/java/io/zeebe/broker/system/configuration/EnvironmentConstants.java]:
```suggestion
  public static final String ENV_STEP_TIMEOUT = "ZEEBE_STEP_TIMEOUT";
```

**Review by deepthidevaki — CHANGES_REQUESTED** [review]
:+1: only a minor change.

File: broker/src/main/java/io/zeebe/broker/system/configuration/BrokerCfg.java (hunk 1 above)
Comment (Zelldon) [reply broker/src/main/java/io/zeebe/broker/system/configuration/BrokerCfg.java]: actually I forgot to do this yes. Thanks.

File: broker/src/main/java/io/zeebe/broker/system/configuration/EnvironmentConstants.java (hunk 2 above)
Comment (Zelldon) [reply broker/src/main/java/io/zeebe/broker/system/configuration/EnvironmentConstants.java]: lol :laughing: :man_facepalming:

**Comment by Zelldon** [comment]
Thanks for the review @deepthidevaki  I updated the branch

**Review by deepthidevaki — APPROVED** [review]

**Pull request merged**